<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Multiple Testing: False discovery rate</title><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="manifest" href="../site.webmanifest"><link rel="mask-icon" href="../safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" content="#ffffff"></head><body>
    <header id="top" class="navbar navbar-expand-md navbar-light bg-white top-nav incubator"><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-6">
      <div class="large-logo">
        <img alt="Carpentries Incubator" src="../assets/images/incubator-logo.svg"><abbr class="badge badge-light" title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught." style="background-color: #FF4955; border-radius: 5px">
          <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link" style="color: #000">
            <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
            Pre-Alpha
          </a>
          <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
        </abbr>
        
      </div>
    </div>
    <div class="selector-container">
      
      
      <div class="dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../False discovery rate.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl navbar-light bg-white bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Carpentries Incubator" src="../assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Multiple Testing
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Multiple Testing
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a class="btn btn-primary" href="../aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Multiple Testing
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 43%" class="percentage">
    43%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 43%" aria-valuenow="43" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../False%20discovery%20rate.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->
      
            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="What%20is%20multiple%20testing.html">1. What is multiple testing</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="Types%20of%20errors.html">2. Types of errors</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="Family-wise%20error%20rate.html">3. Family-wise error rate</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        4. False discovery rate
        </span>
      
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="Pairwise%20comparisons.html">5. Pairwise comparisons</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="Summary.html">6. Summary</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="References.html">7. References</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width resources"><a href="../instructor/aio.html">See all in one page</a>
            

            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">
            
            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/Family-wise%20error%20rate.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/Pairwise%20comparisons.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/Family-wise%20error%20rate.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Family-wise error
        </a>
        <a class="chapter-link float-end" href="../instructor/Pairwise%20comparisons.html" rel="next">
          Next: Pairwise comparisons... 
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>False discovery rate</h1>
        <p>Last updated on 2024-06-18 |
        
        <a href="https://example.com/FIXME-template-source/edit/main/episodes/False%20discovery%20rate.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
        
        
        
        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
        
        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How does correcting for the family-wise error rate (FWER) affect the
number of significant hits in large-scale data, such as RNA-Seq analysis
of 20,000 human genes?</li>
<li>What is the interpretation of a p-value histogram, and how can it be
used to assess the distribution of p-values in multiple testing
scenarios?</li>
<li>How can the Benjamini-Hochberg method be applied to control the
false discovery rate (FDR) in RNA-Seq data, and what are the benefits of
using this method over FWER correction?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Demonstrate how correcting for the family-wise error rate (FWER)
using methods like Bonferroni correction can lead to few or no
significant hits in large-scale testing scenarios.</li>
<li>Introduce the concept of p-value histograms, explain their
interpretation, and illustrate how they can be used to visualize the
distribution of p-values in multiple testing.</li>
<li>Explain the Benjamini-Hochberg method for controlling the false
discovery rate (FDR).</li>
<li>Provide practical examples and R code to apply the
Benjamini-Hochberg method to RNA-Seq data.</li>
<li>Discuss the advantages of controlling the FDR over FWER in the
context of large-scale genomic data.</li>
</ul></div>
</div>
</div>
</div>
</div>
<div class="section level1">
<h1 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a></h1>
<p>In high-throughput experiments like RNA-Seq, we often conduct
thousands of statistical tests simultaneously. This large number of
tests increases the risk of false positives. While controlling the
family-wise error rate (FWER), the probability of making at least one
type I error, is one approach to address false positives, it can be too
conservative, leading to few or no significant results. An alternative
approach is to control the <strong>False Discovery Rate (FDR)</strong>,
the expected proportion of false positives among the rejected
hypotheses, which offers a balance between identifying true positives
and limiting false positives.</p>
<div class="section level2">
<h2 id="example-the-airway-dataset-in-r">Example: The Airway dataset in R<a class="anchor" aria-label="anchor" href="#example-the-airway-dataset-in-r"></a></h2>
<p>The Airway dataset contains gene expression data from a study
investigating the effects of dexamethasone (a corticosteroid medication)
on airway smooth muscle cells. The dataset is part of the airway package
in Bioconductor, a project that provides tools for the analysis and
comprehension of high-throughput genomic data.</p>
<p>In differentially expressed genes (DEGs) analysis, thousands of
statistical tests (one for each gene) are conducted, which increases the
chance of false positives. Using the airway dataset, we can decide to
analyze DEGs between treated and untreated samples. Here, the null
hypothesis would state that there is no difference in gene expression
between the two groups ( dexamethasone treated and untreated). The
<strong>Table 1</strong> below shows the first six rows of the generated
p-values for each gene, the data which, we are going use to see how
using FWER and FDR to controlling for false positive differ.</p>
<table class="table"><caption>Table 1: P_Values for each analysed gene</caption>
<thead><tr class="header"><th align="left">gene</th>
<th align="right">pvalue</th>
</tr></thead><tbody><tr class="odd"><td align="left">ENSG00000000003</td>
<td align="right">0.0286636</td>
</tr><tr class="even"><td align="left">ENSG00000000419</td>
<td align="right">0.0428183</td>
</tr><tr class="odd"><td align="left">ENSG00000000457</td>
<td align="right">0.7874802</td>
</tr><tr class="even"><td align="left">ENSG00000000460</td>
<td align="right">0.6972820</td>
</tr><tr class="odd"><td align="left">ENSG00000000938</td>
<td align="right">0.6215698</td>
</tr><tr class="even"><td align="left">ENSG00000000971</td>
<td align="right">0.0885597</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="the-concept-of-p-value-histograms">The Concept of P-value Histograms<a class="anchor" aria-label="anchor" href="#the-concept-of-p-value-histograms"></a></h2>
<p>A p-value histogram is a graphical representation that displays the
distribution of p-values obtained from multiple hypothesis tests. When
conducting a large number of statistical tests, such as in genome-wide
association studies or RNA-Seq analysis, p-value histograms provide a
visual tool to assess the behavior and distribution of p-values across
all tests.</p>
<p>To create a p-value histogram, we plot the p-values on the x-axis and
the frequency (or count) of those p-values on the y-axis. This
visualization helps in understanding how the p-values are distributed
across the range from 0 to 1.</p>
<p>Here, we will use our generated p-values for each analysed gene to
create a histogram.</p>
<figure><img src="../fig/False%20discovery%20rate-rendered-unnamed-chunk-1-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>We will also find out how many genes were significantly
differentially expressed between the treated and untreated cells at
α=0.05 before we correct for type 1 error.</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>4655 is the number of significant differentially expressed genes before FWER correction</code></pre>
</div>
<div id="discussion1" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge<a class="anchor" aria-label="anchor" href="#discussion1"></a>
</h3>
<div class="callout-content">
<ul><li>Now suppose we reject all tests with a p-value less than α=0.05. How
can we visually determine an estimate of the false discovery proportion
with a plot in r?</li>
</ul></div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alpha</span> <span class="op">=</span> <span class="fl">0.05</span></span>
<span><span class="va">binw</span> <span class="op">=</span><span class="fl">0.01</span></span>
<span><span class="va">pi0</span> <span class="op">=</span> <span class="fl">2</span> <span class="op">*</span> <span class="fu">mean</span><span class="op">(</span><span class="va">gene_pvalues</span><span class="op">$</span><span class="va">pvalue</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">gene_pvalues</span>,</span>
<span>  <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">pvalue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>binwidth <span class="op">=</span> <span class="va">binw</span>, boundary <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">pi0</span> <span class="op">*</span> <span class="va">binw</span> <span class="op">*</span> <span class="fu">nrow</span><span class="op">(</span><span class="va">gene_pvalues</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="va">alpha</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/False%20discovery%20rate-rendered-unnamed-chunk-4-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></div>
</div>
</div>
</div>
<p>Notice that there are many null hypotheses that appear at low
p-values. Therefore, indiscriminately declaring all p-values less than
0.05 as significant will lead to a high number of false discoveries.
Additionally, some true alternative hypotheses may have high p-values,
leading to missed detections (false negatives) that our test cannot
identify.</p>
<div class="section level3">
<h3 id="interpretation-of-p-value-histograms">Interpretation of P-value Histograms<a class="anchor" aria-label="anchor" href="#interpretation-of-p-value-histograms"></a></h3>
<p>A p-value histogram helps identify patterns that might indicate
issues or meaningful findings in our data. If all the null hypotheses
are true (i.e., there is no effect), p-values should follow a uniform
distribution. This means the histogram should show a relatively flat
distribution of p-values across the range from 0 to 1.</p>
<figure><img src="../fig/null_pvalues.png" alt="Figure_8: How p-values would look like if most or all our hypotheses were null" class="figure mx-auto d-block"><div class="figcaption">Figure_8: How p-values would look like if most
or all our hypotheses were null</div>
</figure><p>However, seeing this does not mean they actually are all null. It
means that at most a small percentage of hypotheses are non-null and
could be identified by a correction method. That is why applying an
uncorrected rule like “Accept everything with p-value less than 0.05” is
certain to give you many false discoveries.</p>
<p>If there are true effects present, you would expect to see an excess
of low p-values (close to 0) compared to what would be expected under a
uniform distribution. This indicates potential significant findings. A
large spike near zero can indicate that many tests are finding
significant results, suggesting the presence of true positives.</p>
<figure><img src="../fig/reg_pvalues.png" alt="Figure_9: A peak close to 0 is where our alternative hypotheses live- along with some potential false positives" class="figure mx-auto d-block"><div class="figcaption">Figure_9: A peak close to 0 is where our
alternative hypotheses live- along with some potential false
positives</div>
</figure><p>A right-skewed distribution, with most p-values clustering towards 1,
might suggest that most tests are not finding significant results,
indicating that there may be few true effects in the data.</p>
<figure><img src="../fig/conservative_pvalues.png" alt="Figure_10: Conservative p-values, something is wrong with your test!" class="figure mx-auto d-block"><div class="figcaption">Figure_10: Conservative p-values, something is
wrong with your test!</div>
</figure><p>P-values are designed to be uniform under the null hypothesis.
Therefore, if we observe deviations from this expectation, it should not
lead us to conclude that there are no significant hypotheses. Instead,
it suggests that there may be issues with our testing methodology. For
example, our test assumptions might not align with the actual data
distribution. It is possible that the test assumes a certain data
distribution (like continuous or normal), which doesn’t match the actual
characteristics of our data—such as it being discrete or highly
non-normal.</p>
<p>Some histogram likeshown in <strong>Figure 11</strong> might exhibit
pronounced spikes at p-values near zero and near one.</p>
<figure><img src="../fig/bimodal_pvalues.png" alt="Figure_11: Bimodal p-values" class="figure mx-auto d-block"><div class="figcaption">Figure_11: Bimodal p-values</div>
</figure><p>This behavior often signals potential issues such as p-hacking or
deficiencies in experimental design. For instance, in the context of a
one-tailed test (e.g., assessing whether each gene increases expression
in response to a drug), p-values nearing 1 may indicate significance in
the opposite direction—instances where genes decreased their expression.
If you wish to identify such cases, consider switching to a two-sided
test. Alternatively, if these cases are irrelevant, filtering out
instances where estimates point in that direction could be
effective.</p>
<p>Moreover, in RNA-Seq data, for example, certain genes may show no
reads across all conditions, leading some differential expression tools
to return a p-value of 1. Identifying and filtering out such problematic
cases beforehand can help streamline our analysis without sacrificing
information.”</p>
<p>Understanding the distribution of p-values can inform the choice of
multiple testing correction methods. For example, if there is an excess
of low p-values, controlling the false discovery rate (FDR) might be
more appropriate than controlling the family-wise error rate (FWER).
After applying a multiple testing correction (such as the
Benjamini-Hochberg method), a p-value histogram of the adjusted p-values
can help evaluate the effectiveness of the correction in controlling
false positives.</p>
</div>
</div>
</div>
<div class="section level1">
<h1 id="the-impact-of-correcting-for-the-family-wise-error-rate-fwer-using-bonferroni-correction">The impact of correcting for the family-wise error rate (FWER) using
Bonferroni correction<a class="anchor" aria-label="anchor" href="#the-impact-of-correcting-for-the-family-wise-error-rate-fwer-using-bonferroni-correction"></a></h1>
<p>Now, if we proceed to apply the Bonferroni method to correct the
family-wise error rate (FWER), we can then determine the number of genes
that show significant differential expression after the correction.</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>706 is the number of significant differentially expressed genes after Bonferroni correction for FWER</code></pre>
</div>
<div class="section level2">
<h2 id="interpretation">Interpretation<a class="anchor" aria-label="anchor" href="#interpretation"></a></h2>
<p>Applying the Bonferroni correction in this example results in
<strong>706</strong> differentially expressed genes due to the stringent
threshold, demonstrating how FWER correction can be too conservative in
large-scale testing.</p>
</div>
</div>
<div class="section level1">
<h1 id="controlling-fdr-using-the-benjamini-hochberg-method">Controlling FDR Using the Benjamini-Hochberg Method<a class="anchor" aria-label="anchor" href="#controlling-fdr-using-the-benjamini-hochberg-method"></a></h1>
<p>The Benjamini-Hochberg (BH) method is a statistical procedure used to
control the FDR in multiple hypothesis testing, where the chance of
obtaining false positives increases. The BH method helps control the
proportion of false positives (false discoveries) among the rejected
hypotheses, thus providing a more balanced approach than traditional
methods like the <strong>Bonferroni correction</strong>, which can be
overly conservative. The BH procedure adjusts the p-values from multiple
tests to control the FDR. These adjusted p-values can be compared to a
significance threshold to determine which results are significant.</p>
<div class="section level2">
<h2 id="steps-of-the-benjamini-hochberg-procedure">Steps of the Benjamini-Hochberg Procedure<a class="anchor" aria-label="anchor" href="#steps-of-the-benjamini-hochberg-procedure"></a></h2>
<p>First the observed p-values are arranged in ascending order. Next,
ranks are assigned to the p-values, with the smallest p-value getting
rank 1, the next smallest rank 2, and so on. Then, for each p-value, the
BH critical value is calculated as;</p>
<p><span class="math display">\[\text{BH critical value} = \frac{i}{m}
\times Q\]</span></p>
<p>where, i is the rank, m is the total number of tests, and Q is the
desired FDR level (e.g., 0.05).</p>
<p>After this calculation, the largest p-value that is less than or
equal to its BH critical value is identified. This p-value and all
smaller p-values are considered significant. Optionally, one can adjust
the p-values to reflect the BH correction using software functions.</p>
<p>Now, let us continue with the generated p_values to apply the
Benjamini-Hochberg correction in r, and also plot the adjusted p values
for comparison.</p>
<figure><img src="../fig/False%20discovery%20rate-rendered-unnamed-chunk-6-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>2357 is the number of significant differentially expressed genes that pass the FDR threshold</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="interpretation-1">Interpretation<a class="anchor" aria-label="anchor" href="#interpretation-1"></a></h2>
<p>The Benjamini-Hochberg method controls the FDR, allowing for a
greater number of significant differentially expressed genes
(<strong>2357</strong>) compared to the Bonferroni correction
(<strong>706</strong>). This approach provides a more balanced and
powerful method for identifying true positives in large-scale data.</p>
</div>
</div>
<div class="section level1">
<h1 id="advantages-of-controlling-the-fdr-over-fwer-in-the-context-of-large-scale-genomic-data">Advantages of controlling the FDR over FWER in the context of
large-scale genomic data<a class="anchor" aria-label="anchor" href="#advantages-of-controlling-the-fdr-over-fwer-in-the-context-of-large-scale-genomic-data"></a></h1>
<ul><li><p>Higher statistical power: FDR control allows for higher power
compared to FWER control. In large-scale genomic studies, this means
detecting more true positives while still controlling false discoveries.
Since FDR methods are less stringent than FWER methods like Bonferroni
correction, they balance between controlling false discoveries and
maximizing true positives, reducing false negatives.</p></li>
<li><p>Consistency across studies: By setting a fixed FDR threshold
(e.g., 5%), results are more comparable across different studies and
meta-analyses, enhancing consistency in statistical inference.</p></li>
</ul><p>It is important to note that while FDR is preferred in many genomic
screens due to its ability to handle a large number of hypotheses, FWER
control remains appropriate in scenarios where minimizing any false
positives is critical. Each method has its place depending on the
research context and goals.</p>
<p>NB//: The aim of any multiple hypothesis test correction methods,
such as controlling the Family-Wise Error Rate (FWER) or the False
Discovery Rate (FDR), is to determine an appropriate threshold or cutoff
for declaring statistical significance across multiple tests or
hypotheses.</p>
<div class="section level3">
<h3 id="further-reading">Further reading<a class="anchor" aria-label="anchor" href="#further-reading"></a></h3>
<ul><li><a href="%22http://varianceexplained.org/statistics/interpreting-pvalue-histogram/%22" class="external-link">How
to interpret a p-value histogram</a></li>
</ul><div id="accordionInstructor1" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor1" aria-expanded="false" aria-controls="collapseInstructor1">
  <h3 class="accordion-header" id="headingInstructor1">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div>Instructor Note</h3>
</button>
<div id="collapseInstructor1" class="accordion-collapse collapse" aria-labelledby="headingInstructor1" data-bs-parent="#accordionInstructor1">
<div class="accordion-body">
<p>Inline instructor notes can help inform instructors of timing
challenges associated with the lessons. They appear in the “Instructor
View”</p>
</div>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</div>
</div>



      </div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/Family-wise%20error%20rate.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/Pairwise%20comparisons.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/Family-wise%20error%20rate.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Family-wise error
        </a>
        <a class="chapter-link float-end" href="../instructor/Pairwise%20comparisons.html" rel="next">
          Next: Pairwise comparisons... 
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>
        
        <a href="https://example.com/FIXME-template-source/edit/main/episodes/False%20discovery%20rate.Rmd" class="external-link">Edit on GitHub</a>
        
	
        | <a href="https://example.com/FIXME-template-source/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://example.com/FIXME-template-source/" class="external-link">Source</a></p>
				<p><a href="https://example.com/FIXME-template-source/blob/main/CITATION" class="external-link">Cite</a> | <a href="mailto:carolyn.mukiri@embl.de">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">
        
        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>
        
        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.4" class="external-link">sandpaper (0.16.4)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.5" class="external-link">pegboard (0.7.5)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.2" class="external-link">varnish (1.0.2)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://FIXME-template-source.github.io/NA/instructor/False discovery rate.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries",
  "name": "False discovery rate",
  "creativeWorkStatus": "active",
  "url": "https://FIXME-template-source.github.io/NA/instructor/False discovery rate.html",
  "identifier": "https://FIXME-template-source.github.io/NA/instructor/False discovery rate.html",
  "dateCreated": "2024-04-18",
  "dateModified": "2024-06-18",
  "datePublished": "2024-06-18"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo
    2022-11-07: we have gotten a notification that we have an overage for our
    tracking and I'm pretty sure this has to do with Workbench usage.
    Considering that I am not _currently_ using this tracking because I do not
    yet know how to access the data, I am turning this off for now.
  <script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
    _paq.push(["setDomains", ["*.preview.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
    _paq.push(["setDoNotTrack", true]);
    _paq.push(["disableCookies"]);
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
          var u="https://carpentries.matomo.cloud/";
          _paq.push(['setTrackerUrl', u+'matomo.php']);
          _paq.push(['setSiteId', '1']);
          var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
          g.async=true; g.src='https://cdn.matomo.cloud/carpentries.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
        })();
  </script>
  End Matomo Code --></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

